File: TransCoder_tokenized_test_set_functions/java/PRINT_TRIPLETS_SORTED_ARRAY_FORM_AP.java
static void printAllAPTriplets ( int [ ] arr, int n ) {
  ArrayList < Integer > s = new ArrayList < Integer > ( );
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    for ( int j = i + 1;
    j < n;
    j ++ ) {
      int diff = arr [ j ] - arr [ i ];
      boolean exists = s . contains ( arr [ i ] - diff );
      if ( exists ) System . out . println ( arr [ i ] - diff + " " + arr [ i ] + " " + arr [ j ] );
    }
    s . add ( arr [ i ] );
  }
}

File: TransCoder_tokenized_test_set_functions/python/PRINT_TRIPLETS_SORTED_ARRAY_FORM_AP.py
def printAllAPTriplets ( arr, n ) :
    s = [ ];
    
    for i in range ( 0, n - 1 ) :
        for j in range ( i + 1, n ) :
            diff = arr [ j ] - arr [ i ];
            
            if ( ( arr [ i ] - diff ) in arr ) :
                print ( "{} {} {}" . format ( ( arr [ i ] - diff ), arr [ i ], arr [ j ] ), end = "\n" );
                
    s . append ( arr [ i ] );
